dao/src/main/java/org/thingsboard/server/dao/sql/JpaAbstractDao.java

```diff
@@ -60,21 +85,30 @@ public D save(TenantId tenantId, D domain) {
         }
         setSearchText(entity);
         log.debug("Saving entity {}", entity);
-        if (entity.getUuid() == null) {
+        boolean isNew = entity.getUuid() == null;
+        if (isNew) {
             UUID uuid = Uuids.timeBased();
             entity.setUuid(uuid);
             entity.setCreatedTime(Uuids.unixTimestamp(uuid));
         }
-        entity = getRepository().save(entity);
-        return DaoUtil.getData(entity);
+
+        if (preSaveAction != null) {
+            preSaveAction.accept(entity);
+        }
+        if (TransactionSynchronizationManager.isActualTransactionActive()) {
```

Comment:
As we are going to use save method only, the @Transactional will always present on top

Reply:
The @Transactional is present, but if someone decides to call the save method directly from the dao (like it is now for audit logs), Hibernate will throw an exception, because we cannot use EntityManager without a transaction.

Reply:
I see no problem with annotating 4 methods override
![image](https://github.com/thingsboard/thingsboard/assets/79898499/ec996cf5-8956-42d9-ae23-cbeb18d75166)

At this point. Some transactions should be set or supported. Otherwise - exception - it is desired result